force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -34,19 +34,90 @@ public with sharing class DummyJSONCallout {
      * @param dummyUserId The ID of the user in the external system to fetch data for.
      */
 
-	public static void getDummyJSONUserFromId(String dummyUserId) {
-		// Create HTTP request to send.
-
-		// Set the endpoint URL. Use direct URL or for best practices use Named Credential.
-
-		// Set the HTTP method to GET.
-
-		// Send the HTTP request and get the response.
-
-		// If the HTTP response code is successful, parse the JSON response and update the contact.
-		Contact cont = null;
-		upsert cont DummyJSON_Id__c; //insert/update from the JSON response using the external id (dummyUserId)
-	}
+	 public static void getDummyJSONUserFromId(String dummyUserId) {
+		Http http = new Http();
+		HttpRequest request = new HttpRequest();
+		
+		// Set the endpoint URL
+		String endpoint = 'https://dummyjson.com/users/' + dummyUserId;
+		request.setEndpoint(endpoint);
+		
+		// Set the HTTP method to GET
+		request.setMethod('GET');
+		
+		// Set the header to accept JSON response
+		request.setHeader('Content-Type', 'application/json');
+		
+		try {
+			// Send the request
+			HttpResponse response = http.send(request);
+			
+			// Check if the response was successful (status code 200)
+			if (response.getStatusCode() == 200) {
+				// Parse the JSON response
+				Map<String, Object> responseMap = (Map<String, Object>) JSON.deserializeUntyped(response.getBody());
+				
+				// Extract relevant data from the response
+				String firstName = (String) responseMap.get('firstName');
+				String lastName = (String) responseMap.get('lastName');
+				String email = (String) responseMap.get('email');
+				String phone = (String) responseMap.get('phone');
+				String birthDateStr = (String) responseMap.get('birthDate'); // assuming birthdate is a string in the format 'yyyy-MM-dd'
+				
+				// Convert birthDate to Date if it exists
+				Date birthDate = birthDateStr != null ? Date.valueOf(birthDateStr) : null;
+				
+				// Handle address extraction
+				Map<String, Object> addressMap = (Map<String, Object>) responseMap.get('address');
+				String street = addressMap != null && addressMap.containsKey('address') ? (String) addressMap.get('address') : null;
+				String city = addressMap != null && addressMap.containsKey('city') ? (String) addressMap.get('city') : null;
+				String postalCode = addressMap != null && addressMap.containsKey('postalCode') ? (String) addressMap.get('postalCode') : null;
+				String state = addressMap != null && addressMap.containsKey('state') ? (String) addressMap.get('state') : null;
+				String country = addressMap != null && addressMap.containsKey('country') ? (String) addressMap.get('country') : null;
+				
+				// Query for existing contact using the dummyUserId (DummyJSON_Id__c is an external id field)
+				Contact contact = [SELECT Id, DummyJSON_Id__c FROM Contact WHERE DummyJSON_Id__c = :dummyUserId LIMIT 1];
```

Comment:
You dont need to query for the record first if you are going to do an upsert. In fact this will error if it does not find a record because there will be no results.